src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -153,13 +162,39 @@ public static boolean isProxySet() {
      *
      * @return String: the set proxy address, null if none are set
      */
-    public static String getProxy() {
+    public static String getProxy() throws InvalidProxyException {
         if (!isProxySet()) {
-            return null;
+            throw new InvalidProxyException(PROXY_ISNT_SET);
+        }
+        String proxy = getProgramProperty(PROXY);
+        String[] proxyParts = proxy.split(":");
+        if (proxyParts.length != 2) {
+            throw new InvalidProxyException("Proxy '" + proxy + "' isn't valid. Must contain address and port, without protocol");
+        }
+        try {
+            Integer.parseInt(proxyParts[1]);
+        } catch (NumberFormatException e) {
+            throw new InvalidProxyException("Proxy '" + proxy + "' isn't valid. Must contain address and port, without protocol. Invalid port provided. " + e);
         }
         return getProgramProperty(PROXY);
     }
 
+    public static String getProxyHost() throws InvalidProxyException {
+        if (!isProxySet()) {
+            throw new InvalidProxyException(PROXY_ISNT_SET);
+        }
+        String proxy = getProxy();
+        return proxy.split(":")[0];
+    }
+
+    public static int getProxyPort() throws InvalidProxyException {
+        if (!isProxySet()) {
+            throw new InvalidProxyException(PROXY_ISNT_SET);
+        }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "proxy" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)